src/Symfony/Component/Cache/Adapter/PdoAdapterTrait.php

```diff
@@ -0,0 +1,325 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Adapter;
+
+use Symfony\Component\Cache\Exception\InvalidArgumentException;
+
+/**
+ * @internal
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+trait PdoAdapterTrait
+{
+    /**
+     * @var \PDO|null PDO instance or null when not connected yet
+     */
+    private $pdo;
+
+    /**
+     * @var string|null DSN string or null when lazy connection disabled
+     */
+    private $dsn;
+
+    /**
+     * @var string Database driver
+     */
+    private $driver;
+
+    /**
+     * @var string Table name
+     */
+    private $table = 'cache_items';
+
+    /**
+     * @var string Column for item id
+     */
+    private $idCol = 'item_id';
+
+    /**
+     * @var string Column for item data
+     */
+    private $dataCol = 'item_data';
+
+    /**
+     * @var string Column for expiration timestamp
+     */
+    private $expiryCol = 'item_expiry';
+
+    /**
+     * @var string Username when lazy-connect
+     */
+    private $username = '';
+
+    /**
+     * @var string Password when lazy-connect
+     */
+    private $password = '';
+
+    /**
+     * @var array Connection options when lazy-connect
+     */
+    private $connectionOptions = array();
+
+    private function init($pdoOrDsn, $namespace, $defaultLifetime, array $options)
+    {
+        if (isset($namespace[0]) && preg_match('#[^-+.A-Za-z0-9]#', $namespace, $match)) {
```

Comment:
not sure, less readable, and not perf critical code path anyway (run once code) :)